addons/mail/static/src/models/thread/thread.js

```diff
@@ -658,10 +658,15 @@ function factory(dependencies) {
         /**
          * Marks as read all needaction messages with this thread as origin.
          */
-        async markNeedactionMessagesAsOriginThreadAsRead() {
-            await this.async(() =>
-                this.env.models['mail.message'].markAsRead(this.needactionMessagesAsOriginThread)
-            );
+        async markMessagesAsOriginThreadAsRead() {
+            if (Number.isInteger(this.id)) {
+                await this.async(() => {
+                    this.env.models['mail.message'].markAllAsRead([
+                        ['res_id', '=', this.id],
+                        ['model', '=', this.model]
+                    ]);
```

Comment:
As reminder, the python method needs to be fixed when given a domain, it should return some kind of "channel_ids" so that the JS bus handler does not incorrectly consider this call as "mark all as read" (from inbox) where no domain is provided. Manifested issue related to this is that if there are needaction for other threads/channels than the current one, their needaction counter will incorrectly be reset to 0 in JS.